src/init.cpp

```diff
@@ -853,12 +853,14 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         nLocalServices = ServiceFlags(nLocalServices | NODE_COMPACT_FILTERS);
     }
 
-    // if using block pruning, then disallow txindex and coinstatsindex
+    // if using block pruning, then disallow txindex and coinstatsindex and reindex-chainstate
     if (args.GetIntArg("-prune", 0)) {
         if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX))
             return InitError(_("Prune mode is incompatible with -txindex."));
         if (args.GetBoolArg("-coinstatsindex", DEFAULT_COINSTATSINDEX))
             return InitError(_("Prune mode is incompatible with -coinstatsindex."));
+        if (args.GetBoolArg("-reindex-chainstate", false))
+            return InitError(_("Prune mode is incompatible with -reindex-chainstate. Use full -reindex instead."));
```

Comment:
Might be good to update -prune as well
```diff
-    argsman.AddArg("-prune=<n>", strprintf("Reduce storage requirements by enabling pruning (deleting) of old blocks. This allows the pruneblockchain RPC to be called to delete specific blocks, and enables automatic pruning of old blocks if a target size in MiB is provided. This mode is incompatible with -txindex and -coinstatsindex. "
+    argsman.AddArg("-prune=<n>", strprintf("Reduce storage requirements by enabling pruning (deleting) of old blocks. This allows the pruneblockchain RPC to be called to delete specific blocks, and enables automatic pruning of old blocks if a target size in MiB is provided. This mode is incompatible with -txindex, -coinstatsindex, and -reindex-chainstate. "
```

Reply:
Hmm, can you imagine a situation where this would help anyone? In contrast to the index options, `-reindex-chainstate` doesn't enable some additional optional feature, but is a one-time measure to recover when from a corrupted chainstate. So I would think at the point where you chose to enable pruning, you probably don't care or need to know about `reindex-chainstate` at all. Or is the general strategy to list all interactions for completeness?